includes/multi-currency/class-wc-payments-multi-currency.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Class WC_Payments_Multi_Currency
+ *
+ * @package WooCommerce\Payments
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that controls Multi Currency functionality.
+ */
+class WC_Payments_Multi_Currency {
+
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var WC_Payments_Multi_Currency
+	 */
+	protected static $instance = null;
+
+	/**
+	 * The available currencies.
+	 *
+	 * @var array
+	 */
+	public $available_currencies;
+
+	/**
+	 * The default currency.
+	 *
+	 * @var object
+	 */
+	public $default_currency;
+
+	/**
+	 * The enabled currencies.
+	 *
+	 * @var array
+	 */
+	public $enabled_currencies;
+
+	/**
+	 * Main WC_Payments_Multi_Currency Instance.
+	 *
+	 * Ensures only one instance of WC_Payments_Multi_Currency is loaded or can be loaded.
+	 *
+	 * @static
+	 * @return WC_Payments_Multi_Currency - Main instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->init();
+	}
+
+	/**
+	 * Init.
+	 */
+	public function init() {
+		include_once WCPAY_ABSPATH . 'includes/multi-currency/class-wc-payments-multi-currency-currency.php';
+
+		$this->id = 'wcpay_multi_currency';
+		$this->get_available_currencies();
+		$this->get_default_currency();
+		$this->get_enabled_currencies();
+	}
+
+	/**
+	 * Gets the currencies available.
+	 *
+	 * @return array Array of WC_Payments_Multi_Currency_Currency objects.
+	 */
+	public function get_available_currencies() {
+		if ( isset( $this->available_currencies ) ) {
+			return $this->available_currencies;
+		}
+
+		// TODO: Mock currencies to be replaced by stored values from API call.
```

Comment:
Nitpick: perhaps we should extract the `$mock_currencies` from here and create a static method (something like `create_mock_currencies` which handles creating the mock currencies, and then this function could just call `create_mock_currencies` and set the value of `$this->available_currencies` to what gets returned.

Reply:
I abstracted this a bit in https://github.com/Automattic/woocommerce-payments/pull/1795/commits/cdb969523e396de384ea946505bdcbb3a4b144b9 , but I didn't want to go too far into creating mock currencies. For now I just moved the currencies out to another function that returns the array that resembles what is returned from the exchange service.

Reply:
Yeah, that makes sense and just moving it out to another function is perfect for now I think!